File: palindrome/palindrome.js
function palindrome(value) {
  const valueWithNoSpaces = value.replace(/\s/g, '');

  if (value === value.split('').reverse().join('')) {
    return 'It`s palindrome';
  } else if (valueWithNoSpaces === valueWithNoSpaces.split('').reverse().join('')) {
    return 'With no spaces, it`s palindrome';
  }

  return 'It`s not palindrome';
}

File: palindrome/palindrome.rb
def palindrome(string)
  string_with_no_spaces = string.gsub(' ', '')

  if string == string.reverse
    'It`s palindrome'
  elsif string_with_no_spaces == string_with_no_spaces.reverse
    'With no spaces, it`s palindrome'
  else
    'It`s not palindrome'
  end
end
